mp/src/game/shared/momentum/util/engine_patch.cpp

```diff
@@ -0,0 +1,260 @@
+    //    "xxxxx????",
+    //    5,
+    //    PATCH_IMMEDIATE,
+    //    "\x00\x40\x00\x00"
+    //}
+};
+#elif __linux__
+CEnginePatch g_EnginePatches[] =
+{
+    // Prevent the culling of skyboxes at high FOVs
+    // https://github.com/VSES/SourceEngine2007/blob/master/se2007/engine/gl_warp.cpp#L315
+    // TODO: Use a value derived from FOV instead
+    {
+        "SkyboxCulling",
+        "\xF3\x0F\x59\x0D\x00\x00\x00\x00\xF3\x0F\x58\xC2\xF3\x0F\x58\xC1\xF3\x0F\x10\x0D",
+        "xxxx????xxxxxxxxxxxx",
+        20,
+        PATCH_REFERENCE,
+        -1.0f
+    },
+    // Example patch: Trigger "Map has too many brushes" error at 16384 brushes instead of 8192
+    //{
+    //    "BrushLimit",
+    //    "\xBE\x00\x00\x00\x00\xF7\xE6\x89\xD6\xC1\xEE\x03\x81\xFE",
+    //    "x????xxxxxxxxx",
+    //    14,
+    //    PATCH_IMMEDIATE,
+    //    16384
+    //},
+    // The same patch as above but using a pure hex value
+    //{
+    //    "BrushLimitHex",
+    //    "\xBE\x00\x00\x00\x00\xF7\xE6\x89\xD6\xC1\xEE\x03\x81\xFE",
+    //    "x????xxxxxxxxx",
+    //    14,
+    //    PATCH_IMMEDIATE,
+    //    "\x00\x40\x00\x00"
+    //}
+};
+#endif //_WIN32
+
+void CEnginePatch::ApplyAll()
+{
+#if !defined (OSX) // No OSX patches
+    int err;
+
+    for (int i = 0; i < sizeof(g_EnginePatches) / sizeof(*g_EnginePatches); i++)
+    {
+        err = g_EnginePatches[i].ApplyPatch();
+        switch (err)
+        {
+        case PATCH_SUCCESS:
+            DevLog("Engine patch \"%s\" applied successfully\n", g_EnginePatches[i].GetName());
+            break;
+        case PATCH_MEMPROTECT_FAIL:
+            Warning("Engine patch \"%s\" FAILED: Could not override memory protection\n", g_EnginePatches[i].GetName());
+            break;
+        case PATCH_INVALID_SIGNATURE:
+            Warning("Engine patch \"%s\" FAILED: Could not find signature\n", g_EnginePatches[i].GetName());
+            break;
+        case PATCH_NO_VALUE:
+            Warning("Engine patch \"%s\" FAILED: No value provided\n", g_EnginePatches[i].GetName());
+        }
+    }
+#endif // (OSX)
+}
+
+// Apply a patch to engine memory
+int CEnginePatch::ApplyPatch()
+{
+    if (!m_pPatch)
+        return PATCH_NO_VALUE;
+
+    void* addr = CEngineBinary::FindPattern(m_pSignature, m_pMask, m_iOffset);
+
+    if (addr)
+    {
+        auto pMemory = m_bImmediate ? (uintptr_t*)addr : *reinterpret_cast<uintptr_t**>(addr);
+
+        // Memory is write-protected so it needs to be lifted before the patch is applied
+#ifdef _WIN32
+        // https://docs.microsoft.com/en-us/windows/win32/memory/memory-protection-constants
+        unsigned long iOldProtection;
+        unsigned long iNewProtection = PAGE_EXECUTE_READWRITE;
+#else //POSIX
+        // http://man7.org/linux/man-pages/man2/mprotect.2.html
+        unsigned long iOldProtection = PROT_READ | PROT_EXEC;
+        unsigned long iNewProtection = PROT_READ | PROT_WRITE | PROT_EXEC;
+#endif //_WIN32
```

Comment:
I'd rather if you just added your own flags and converted to the OS specific flags in the `SetMemoryProtection` function, that'll let you get rid of all the OS dependent code in this function